nitransforms/base.py

```diff
@@ -160,21 +216,27 @@ def resample(self, moving, order=3, mode='constant', cval=0.0, prefilter=True,
 
         Returns
         -------
-        moved_image : `spatialimage`
-            The moving imaged after resampling to reference space.
+        resampled : `spatialimage` or ndarray
+            The data imaged after resampling to reference space.
 
         """
-        if isinstance(moving, str):
-            moving = load(moving)
+        if reference is not None and isinstance(reference, (str, Path)):
+            reference = load(reference)
+
+        _ref = self.reference if reference is None \
+            else SpatialReference.factory(reference)
 
-        moving_data = np.asanyarray(moving.dataobj)
-        output_dtype = output_dtype or moving_data.dtype
-        targets = ImageGrid(moving).index(
-            _as_homogeneous(self.map(self.reference.ndcoords.T),
-                            dim=self.reference.ndim))
+        if isinstance(spatialimage, str):
```

Comment:
since we're allowing reference to be a `Path`
```suggestion
        if isinstance(spatialimage, (str, Path)):
```